cms/models/placeholdermodel.py

```diff
@@ -429,5 +431,48 @@ def mark_as_dirty(self, language, clear_cache=True):
         elif attached_model is StaticPlaceholder:
             StaticPlaceholder.objects.filter(draft=self).update(dirty=True)
 
+    def get_vary_cache_on(self, request):
+        """
+        Returns a list of VARY headers.
+        """
+        def inner_plugin_iterator(lang):
+            """See note in get_cache_expiration.inner_plugin_iterator()."""
+            if hasattr(self, '_all_plugins_cache'):
+                for instance in self._all_plugins_cache:
+                    plugin = instance.get_plugin_class_instance()
+                    yield instance, plugin
+            else:
+                for plugin_item in self.get_plugins(lang):
+                    yield plugin_item.get_plugin_instance()
+
+        if not self.cache_placeholder or not get_cms_setting('PLUGIN_CACHE'):
+            return list()
+
+        vary_list = list()
```

Comment:
please use a set() to avoid all the checks if an item is in or not.
I recommend just storing them as lower cased values because anyways the check is insensitive.